### PR TITLE
Move DynamicCodeSupport default into aot targets

### PR DESCRIPTION
**Pull request by sbomer**
I think this default setting should live alongside the other feature switch defaults in the aot targets. Currently it's defined in the SDK. Once this change flows to SDK I'll remove it there: https://github.com/dotnet/sdk/pull/33040

### CODE REVIEW

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets [hunk 1]
```diff
@@ -43,6 +43,7 @@ The .NET Foundation licenses this file to you under the MIT license.
     <UseSystemResourceKeys Condition="$(IlcDisableReflection) == 'true'">true</UseSystemResourceKeys>
     <EventSourceSupport Condition="$(IlcDisableReflection) == 'true'">false</EventSourceSupport>
     <EventSourceSupport Condition="$(EventSourceSupport) == ''">false</EventSourceSupport>
+    <DynamicCodeSupport Condition="'$(DynamicCodeSupport)' == ''">false</DynamicCodeSupport>
```
Comment (eerhardt) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets:46]: I assume this .targets file gets imported even when I'm not actually publishing, but just "building" with `PublishAot=true` set, correct?

Comment (eerhardt) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets:46]: Is there any way to test these things in `dotnet/runtime`?

Comment (MichalStrehovsky) [reply src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets:46]: We compile all NativeAOT tests in dotnet/runtime using this target so we could check for this in NativeAOT (look for the appcontext switch at runtime). But it won't cover CoreCLR so we could still have a bug that this is only set in Publish and not know about it. Not sure it's worth the trouble just for that.

File: src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets [hunk 2]
```diff
@@ -43,6 +43,7 @@ The .NET Foundation licenses this file to you under the MIT license.
     <UseSystemResourceKeys Condition="$(IlcDisableReflection) == 'true'">true</UseSystemResourceKeys>
```
Comment (MichalStrehovsky) [inline src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets:43]: Could you delete the PublishTrimmed on line 28 while you're on it? I noticed in your SDK PR that we set this to true on the SDK side so this should not be needed.

**Review by MichalStrehovsky — APPROVED** [review]

**Pull request merged**